### ISSUE TITLE
Error when specifying Deck Sort Methods

### ISSUE THREAD

**Issue by xiaoxiae** [Stvad/CrowdAnki#98]
Hello,

I have a particular CrowdAnki setup, where a snapshot of the deck is automatically created on closing Anki and commited to Git. This has worked wonders for the past year, but when opening it today, I noticed that the order of the items in JSON keeps changing as the snapshots are created.

This is obviously a huge problem, so I tried to remedy this by using TODO in Tools > Add-ons > CrowdAnki > Deck Sort Methods. However, changing the text in any way results in the following error:

```
Caught exception:
Traceback (most recent call last):
  File "/home/xiaoxiae/Documents/Anki/addons21/1788670778/config/config_dialog.py", line 77, in changed_textedit_deck_sort_methods
    self.config.export_note_sort_methods = self.string_cs_to_list(
AttributeError: 'ConfigDialog' object has no attribute 'string_cs_to_list'
```

Am I doing something wrong?

**Comment by xiaoxiae** [comment]
It seems that removing `self.` before `string_cs_to_list` works (since `string_cs_to_list` is a function imported from `..utils.utils`), but it seems that it has no effect and the attributes are still sorted quite randomly.

**Comment by aplaice** [comment]
Thanks very much for diagnosing the miscalling of `string_cs_to_list` — it indeed fixes the crash!

However, with the fix, I can't reproduce the issue of the snapshots not being sorted.

Both `field1` and `guid` as the "deck sort method" seemed to work as expected. I tried it with Anki 2.1.26, 2.1.28 and 2.1.34.

Which sort method and which Anki version are you using?

(Thanks for reporting!)

**Comment by xiaoxiae** [comment]
I tried using both to no avail, but maybe I just don't understand what it's supposed to do. I expect there to be no diff when I create two snapshots, one after another, without changing my Anki deck. Using `guid` and `field1` doesn't seem to do this.

**Comment by aplaice** [comment]
Yes, there should be a diff, when changing the sorting method! (Except if your deck happens to be already sorted by both fields, but that's obviously unlikely enough to be effectively impossible.)

**Comment by ohare93** [comment]
@xiaoxiae do you mind sharing what you are exporting? Just the json is fine. Then we can inform you whether the export happened successfully, and things were sorted by guid then field1 👍 A screenshot of your config settings window just before successfully saving would also go a long way. Thanks mate.

**Comment by xiaoxiae** [comment]
I don't mind at all. Using Anki 2.1.26 and the following config:
```
{"name": "CrowdAnki JSON exportimport Edit history Collaborate on deck creation", "mod": 1593991615, "min_point_version": 26, "max_point_version": 26, "branch_index": 1, "config": {"automated_snapshot": true, "export_note_sort_methods": ["guid", "field1"], "export_notes_reverse_order": false, "import_notes_ignore_deck_movement": false, "snapshot_path": "/home/xiaoxiae/Documents/Anki/addons21/1788670778/user_files", "snapshot_root_decks": ["\u0160kola"], "export_create_deck_subdirectory": true}, "disabled": false, "conflicts": []}
```

When saving two times in succession, without changing the deck, the following JSON snapshot files are produced (uploaded as `.txt` because GitHub):

[deck1.txt](https://github.com/Stvad/CrowdAnki/files/5300150/deck1.txt)
[deck2.txt](https://github.com/Stvad/CrowdAnki/files/5300151/deck2.txt)

**Comment by ohare93** [comment]
Ahhh I see. Your *notes* are being correctly sorted, as was the intention of the Deck Sort method. However the thing that seems to be apparently random is the ordering of *the subdecks themselves* in the json :thinking: This is very strange behaviour that should be rectified (just ordered by name of the deck, as Anki already does it, I'd say) but it has nothing to do with the Deck Sort method. You can get around this issue by exporting each of the decks individually :+1: that should result in no more jumping around, and the files will remain consistent, but do let me know if they do not.

I implemented the Deck Sort option, so I should be able to fix this no problem. I'll get around to fixing this at some point and will ping here when it is done :+1:

**Comment by xiaoxiae** [comment]
Ah, that makes sense. Exporting individually would be a little tedious though, since I would have to manually specify the roots of the snapshots, no? Either way, thanks a lot for the explanation and help!

**Comment by ohare93** [comment]
@xiaoxiae I have fixed the Subdeck ordering issue, see #103 for the simple change fix.
